apps/user-profile/src/components/common/Stepper/Stepper.js

```diff
@@ -0,0 +1,168 @@
+import { useEffect, useState } from 'react';
+import {
+  StepperHeader,
+  StepperBody,
+  StepperFooter,
+  ProgressBarContainer,
+  Bar,
+  EmptyBar,
+} from './StyledStepper';
+import { Button } from '@devlaunchers/components/components/atoms';
+
+const stepsMockData = [
+  {
+    component: <p>Step 1</p>,
+    config: {
+      buttons: {
+        next: {
+          label: 'Next',
+          useDefaultOnClick: true,
+          onClick: () => {
+            console.log('..some function ');
+          },
+        },
+        back: {
+          label: 'Back',
+          useDefaultOnClick: true,
+          // useDefaultHide: true,
+          onClick: () => {
+            console.log('..some function ');
+          },
+        },
+      },
+    },
+  },
+  {
+    component: <p>Step 2</p>,
+    config: {
+      buttons: {
+        next: {
+          label: 'Next',
+          useDefaultOnClick: true,
+          onClick: () => {
+            console.log('..some function ');
+          },
+        },
+        back: {
+          label: 'Back',
+          useDefaultOnClick: true,
+          onClick: () => {
+            console.log('..some function ');
+          },
+        },
+      },
+    },
+  },
+  {
+    component: <p> Step 3</p>,
+    config: {
+      buttons: {
+        next: {
+          label: 'Finish',
+          useDefaultOnClick: false,
+          onClick: () => {
+            Router.push('./users-me');
+          },
+        },
+        back: {
+          label: 'Back',
+          useDefaultOnClick: true,
+          onClick: () => {
+            console.log('..some function ');
+          },
+        },
+      },
+    },
+  },
+];
+
+/**
+ * @description This is a stepper component that is used for the user onboarding
+ */
+export default function Stepper({ steps = stepsMockData }) {
+  const [stepsData, setStepsData] = useState(steps);
+  const [index, setIndex] = useState(0);
+
+  const [activeComponent, setActiveComponent] = useState(
+    stepsData[index].component
+  );
+  const [activeConfig, setActiveConfig] = useState(stepsData[index].config);
+  const [buttonConfig, setButtonConfig] = useState(
+    stepsData[index].config.buttons
+  );
+
+  const backOnClickHandler = () => {
+    if (
+      (buttonConfig.back.useDefaultOnClick === undefined) |
```

Comment:
This can be simplified to `if(buttonConfig?.back?.useDefaultOnClick) {...}` if it's undefined it's going to be false.

Reply:
the `?.` is known as Optional Chaining operator, it returns undefined, if the properties does not exist in the object